pkg/sink/sink.go

```diff
@@ -0,0 +1,131 @@
+/*
+Copyright 2019 The Tekton Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package sink
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"path"
+	"strings"
+
+	triggersclientset "github.com/tektoncd/triggers/pkg/client/clientset/versioned"
+	"github.com/tektoncd/triggers/pkg/template"
+	"github.com/tidwall/gjson"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	kubeclientset "k8s.io/client-go/kubernetes"
+)
+
+type Resource struct {
+	TriggersClient         *triggersclientset.Clientset
+	KubeClient             *kubeclientset.Clientset
+	EventListenerName      string
+	EventListenerNamespace string
+}
+
+func (r Resource) HandleEvent(response http.ResponseWriter, request *http.Request) {
+	el, err := r.TriggersClient.TektonV1alpha1().EventListeners(r.EventListenerNamespace).Get(r.EventListenerName, metav1.GetOptions{})
+	if err != nil {
+		log.Printf("Error getting EventListener %s in Namespace %s: %s", r.EventListenerName, r.EventListenerNamespace, err)
+		return
+	}
+
+	event, err := ioutil.ReadAll(request.Body)
+	if err != nil {
+		log.Printf("Error reading event body: %s", err)
+		return
+	}
+	log.Printf("Event: %s", string(event))
+
+	for _, tbRef := range el.Spec.TriggerBindingRefs {
+		tbNamespace := tbRef.Namespace
+		if tbNamespace == "" {
+			tbNamespace = r.EventListenerNamespace
+		}
+		tb, err := r.TriggersClient.TektonV1alpha1().TriggerBindings(tbNamespace).Get(tbRef.Name, metav1.GetOptions{})
+		if err != nil {
+			log.Printf("Error getting TriggerBinding %s in Namespace %s: %s", tbRef.Name, tbNamespace, err)
```

Comment:
I know we're logging debug messages a lot at this stage, but this line should eventually be `errorf`right? We could also propagate errors up and check if nil in main.

Reply:
Unfortunately, I think the `log` package only has `Printf` and `Fatalf`, there is no `Errorf`.

For propagating the errors, I think I propagate all of the errors up to the `HandleEvent` function in `sink.go`. However, this is a handler function, and is not invoked directly in `main.go`: `http.HandleFunc("/", r.HandleEvent)`, so I don't think I can propogate errors further up into `main.go`?

Reply:
Ah, makes sense :+1: